apps/tangle-dapp/src/containers/restaking/WithdrawModal.tsx

```diff
@@ -38,18 +42,22 @@ const WithdrawModal = ({
 }: Props) => {
   const { vaults } = useRestakeContext();
 
-  // Aggregate the delegations based on the operator account id and asset id
-  const deposits = useMemo(() => {
+  const availableForWithdrawal = useMemo(() => {
     if (!delegatorInfo?.deposits) {
       return [];
     }
 
-    return Object.entries(delegatorInfo.deposits).map(
-      ([assetId, { amount }]) => ({
-        assetId: assertRestakeAssetId(assetId),
-        amount,
-      }),
-    );
+    return Object.entries(delegatorInfo.deposits).map(([assetIdString]) => {
+      const availableForWithdrawal = calculateRestakeAvailableBalance(
```

Comment:
The backend has a field called [`delegated_amount`](https://github.com/tangle-network/tangle/blob/59c719c14f6f3834546eb22c7b61b119dfd1989b/pallets/multi-asset-delegation/src/types/delegator.rs#L220-L221) that indicates the total delegated amount, which can be used to calculate the available balance. 

This issue has been addressed in PR https://github.com/tangle-network/dapp/pull/2788.